src/formats/smilesformat.cpp

```diff
@@ -2364,6 +2364,57 @@ namespace OpenBabel {
     return(idx);
   }
 
+  // Is this dbl bond in a ring that is of size <= IMPLICIT_CIS_RING_SIZE?
+  //
+  // A bounded BFS from one side of a bond trying to find the other side.
+  //   - The required queue is implemented using a std::vector for efficiency
+  //   - Note that items are never popped from the queue, I just move a pointer
+  //     along to mark the left hand side
+  //   - A potential improvement would be to BFS from both sides at the same time
+  static bool IsNotInRingOfSizeThatImpliesCis(OBMol *mol, OBBond *bond)
```

Comment:
I'd actually prefer if we can make this a general method, e.g., OBMolAtomBFSIter. Many of the calls to SSSR could be replaced by an iterator like this (e.g., aromaticity)

Reply:
Actually I almost did this. Well I did it, and then reverted, to be honest. But I can put it back. My only concern is that while this is efficient for a single atom, I think a different approach would be better if you are interested in identifying all atoms in smallish rings (I had a discussion on this with @johnmay).

But for sure, there's no harm in making the method available.